src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -0,0 +1,465 @@
+     * layer depends.
+     */
+    public RTPEncodingImpl(
+        MediaStreamTrackImpl track, int idx,
+        long primarySSRC, long rtxSSRC,
+        int temporalId,
+        RTPEncodingImpl[] dependencyEncodings)
+    {
+        this.primarySSRC = primarySSRC;
+        this.rtxSSRC = rtxSSRC;
+        this.track = track;
+        this.idx = idx;
+        this.temporalId = temporalId;
+        this.dependencyEncodings = dependencyEncodings;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getPrimarySSRC()
+    {
+        return primarySSRC;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getRTXSSRC()
+    {
+        return rtxSSRC;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isActive()
+    {
+        return active;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString()
+    {
+        return "subjective_quality=" + idx +
+            ",primary_ssrc=" + getPrimarySSRC() +
+            ",rtx_ssrc=" + getRTXSSRC() +
+            ",temporal_id=" + temporalId +
+            ",active=" + active +
+            ",last_stable_bitrate_bps=" + lastStableBitrateBps;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public MediaStreamTrackImpl getMediaStreamTrack()
+    {
+        return track;
+    }
+
+    /**
+     * Gets the subjective quality index of this instance.
+     *
+     * @return the subjective quality index of this instance.
+     */
+    int getIndex()
+    {
+        return idx;
+    }
+
+    /**
+     * Returns a boolean that indicates whether or not this
+     * {@link RTPEncodingImpl} depends on the subjective quality index that is
+     * passed as an argument.
+     *
+     * @param idx the index of this instance in the track encodings array.
+     * @return true if this {@link RTPEncodingImpl} depends on the subjective
+     * quality index that is passed as an argument, false otherwise.
+     */
+    boolean requires(int idx)
+    {
+        if (idx < 0)
+        {
+            return false;
+        }
+
+        if (idx == this.idx)
+        {
+            return true;
+        }
+
+
+        boolean requires = false;
+
+        if (!ArrayUtils.isNullOrEmpty(dependencyEncodings))
+        {
+            for (RTPEncodingImpl enc : dependencyEncodings)
+            {
+                if (enc.requires(idx))
+                {
+                    requires = true;
+                    break;
+                }
+            }
+        }
+
+        return requires;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean matches(byte[] buf, int off, int len)
+    {
+        long ssrc = RawPacket.getSSRCAsLong(buf, off, len);
+
+        if (primarySSRC != ssrc && rtxSSRC != ssrc)
+        {
+            return false;
+        }
+
+        int tid = track.getMediaStreamTrackReceiver()
+            .getStream().getTemporalID(buf, off, len);
```

Comment:
I'm sorry, I should've clarified that I'm planing to move the RTX code in the next couple of days.